server/game/player.js

```diff
@@ -202,6 +202,9 @@ class Player extends GameObject {
             });
             this.addTopCardOfDeckVisibleMessage();
         }
+        if (shuffledDiscardIntoDeck) {
+            this.game.raiseEvent('onDiscardShuffledIntoDeck', { player: this });
```

Comment:
I don't think this should be a separate event and should probably be a parameter of the shuffle event

Reply:
@cryogen I had two issues with doing it this way:
1. The original shuffle event is emitted _before_ the actual event happens, for some reason.  This trigger needs to happen after.
2. The other event uses `emitEvent`, which seems to make it impossible to listen for because it does some weird thing where it appends `:reaction` to the event name.  `raiseEvent` seems to take care of that.

All that said, nothing seems to use the `onDeckShuffled` aside from tests, at least on the server side, so maybe we could just replace that event completely.